src/node_nanomsg.cc

```diff
@@ -229,30 +233,14 @@ NAN_METHOD(PollSendSocket) {
   }
 }
 
-NAN_METHOD(PollReceiveSocket) {
-  int s = Nan::To<int>(info[0]).FromJust();
-  Nan::Callback *callback = new Nan::Callback(info[1].As<Function>());
-
-  nanomsg_socket_t *context;
-  size_t siz = sizeof(uv_os_sock_t);
-
-  context = reinterpret_cast<nanomsg_socket_t *>(calloc(1, sizeof *context));
-  context->poll_handle.data = context;
-  context->callback = callback;
-  nn_getsockopt(s, NN_SOL_SOCKET, NN_RCVFD, &context->sockfd, &siz);
-
-  if (context->sockfd != 0) {
-    uv_poll_init_socket(uv_default_loop(), &context->poll_handle,
-                        context->sockfd);
-    uv_poll_start(&context->poll_handle, UV_READABLE, NanomsgReadable);
-    info.GetReturnValue().Set(WrapPointer(context, 8));
-  }
+static void close_cb(uv_handle_t *handle) {
+  NanoMsgPollCtx *context = reinterpret_cast<NanoMsgPollCtx*>(handle->data);
+  delete context;
 }
 
 NAN_METHOD(PollStop) {
-  nanomsg_socket_t *context = UnwrapPointer<nanomsg_socket_t *>(info[0]);
-  int r = uv_poll_stop(&context->poll_handle);
-  info.GetReturnValue().Set(Nan::New<Number>(r));
+  NanoMsgPollCtx *context = UnwrapPointer<NanoMsgPollCtx*>(info[0]);
+  uv_close(reinterpret_cast<uv_handle_t*>(&context->poll_handle), close_cb);
```

Comment:
../src/node_nanomsg.cc:250:12: error: static_cast from 'uv_poll_t *' (aka 'uv_poll_s *')
      to 'uv_handle_t *' (aka 'uv_handle_s *'), which are not related by inheritance, is
      not allowed
  uv_close(static_cast<uv_handle_t*>(&context->poll_handle), close_cb);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~